src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1136,9 +1140,12 @@
                                     :end_time               end_time
                                     :receive_time           (to-timestamp received-timestamp)
                                     :environment_id         (ensure-environment environment)
-                                    :status_id              (ensure-status status)}))]
+                                    :status_id              (ensure-status status)}))
+                   assoc-ids #(assoc %
+                                     :report_id id
+                                     :certname_id (certname-id certname))]
                (->> resource_events
-                    (map (comp convert-containment-path #(assoc % :report_id id)))
+                    (map (comp convert-containment-path assoc-ids))
                     (apply sql/insert-records :resource_events))
                (when update-latest-report?
                  (update-latest-report! certname)))))))
```

Comment:
there's no certname_id in the reports table as a result of this patch. I'm pretty sure I'd support that change but I think it would hit enough places to warrant its own ticket.